### ISSUE TITLE
Discuss: Define a list of scopes for git commit messages

### ISSUE THREAD

**Issue by shane-tomlinson** [mozilla/fxa#1062]
I saw this mentioned by @bbangert yesterday that we do not have a list of defined scopes for git commit messages. We [define the types](https://github.com/mozilla/fxa/blob/b2e78d0231abaf2da864e575293a05c1004e4586/CONTRIBUTING.md#git-commit-guidelines) but not the scopes.

Some quick possibilities, though other's are usually better at this stuff than I am:

* package name, e.g., `fix(content): blah blah`
* package-name:type, e.g., `fix(content:style): blah blah`
* type, e.g., `fix(style): blah blah`

**Comment by lmorchard** [comment]
This could be related to #759 also if we want to enforce / lint the messages.

Would it help to maybe chew through the commit history so far and get a list of what we've been using as a starting point?

**Comment by philbooth** [comment]
Also note that whatever whatever weird and wonderful convention we come up with here, we need to fix the release script to handle the new format(s). Recently there has been some experimentation going on in people's commit messages but the release script was not updated to handle them, so the changelogs look kind of broken in places. In the meantime, if people could stick to the old conventions for now it would be helpful.

**Comment by clouserw** [comment]
Angular, which we base our commit message format off of, uses [this guidance](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#scope) for scopes:

> The scope could be anything specifying place of the commit change. For example $location, $browser, $compile, $rootScope, ngHref, ngClick, ngView, etc...
> 
> You can use * when the change affects more than a single scope.

My gut feeling is that defining scopes beyond that will tip the balance to be burdensome, and at some point you'll need an `other` category anyway, unless the list is so long it's unwieldy anyway.  I think flexibility here will keep the barrier to contributing low (so we can raise it in other areas that are critical).

**Comment by philbooth** [comment]
I think what *might* be useful, rather than a list of allowed scopes, is some rules about things that shouldn't be scopes. Although possibly it depends on your point of view.

Right now, I see a lot of people using package names as the scope in the commit message. I think that's suboptimal because it makes the generated changelogs tautological. Each changelog is already scoped to a particular package, so if parsing the package name from that scope field and including it in the changelog, it's redundant and something of a wasted opportunity.

Example, in case it's not clear:

https://github.com/mozilla/fxa/blob/v1.144.0/packages/fxa-payments-server/CHANGELOG.md#11440

**Comment by bbangert** [comment]
@philbooth Agreed, if we use a scope of a package name, is it 'event-broker', 'event broker', or 'fxa-event-broker'? Just having a rule for that would be great.
